lib/components/Charts/PieChart/index.tsx

```diff
@@ -0,0 +1,49 @@
+import { ChartContainer, ChartTooltip, ChartTooltipContent } from "@/ui/chart"
+import { ForwardedRef } from "react"
+import { Pie, PieChart as PieChartPrimitive } from "recharts"
+import { autoColor } from "../utils/colors"
+import { fixedForwardRef } from "../utils/forwardRef"
+import { ChartConfig, InferChartKeys } from "../utils/types"
+
+export type PieChartItem = {
+  label: string
+  value: number
+  color?: string
+}
+
+export type PieChartProps<
+  DataConfig extends ChartConfig = ChartConfig,
+  Keys extends string = InferChartKeys<DataConfig>,
+> = {
+  dataConfig: ChartConfig<Keys>
+  data: PieChartItem[]
+  donutPieChart?: boolean
```

Comment:
Do we really want to support both? I'm in favor of starting with fewer options and deciding later.

Reply:
I think we will need it, we have some use cases for using donut pie chart at some places.